### PR TITLE
ci(json-schema-check): improve readibility and output helpful debug info

### PR DESCRIPTION
**Pull request by xmfcx**
## Description

- This PR makes it easier to find the source of errors in the checks.
- Also makes it more clear on what exact files are checked against which exact schemas.
- Also converts the bash one-liner to a python script for maintainability.

## Tests performed

Temporarily created a tag named `vTempTestJson` to test it from Universe.

### Before
- https://github.com/autowarefoundation/autoware.universe/actions/runs/7979187696/job/21793290925#step:3:42

Screenshot from 2024-02-21 02-39-21

### After
- https://github.com/autowarefoundation/autoware.universe/actions/runs/7981506848/job/21793322599#step:3:43

Screenshot from 2024-02-21 02-38-49

## Effects on system behavior

Colorful outputs!

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by xmfcx** [comment]
Sorry @VRichardJP for tagging you, I saw the name Vincent in the git blame and added you. But it was Ambroise Vincent :D

**Comment by xmfcx** [comment]
@ambroise-arm this is ready for review :bow:

File: json-schema-check/validate_json_schemas.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+                colorama.Fore.BLUE + f'{schema_file}' +
+                colorama.Style.RESET_ALL,
+                end=' '
+            )
+            result = subprocess.run(['check-jsonschema', '--schemafile', schema_file, config_file], capture_output=True)
+            if result.returncode != 0:
+                print(colorama.Fore.RED + '❌ Failed')
+                for line in result.stdout.decode('utf-8').split('\n'):
+                    if line:
+                        print(colorama.Fore.RED + line)
```
Comment (ambroise-arm) [inline json-schema-check/validate_json_schemas.py:36]: Maybe we could indent those a little? If it makes sense, then probably the "no configuration files found" should be indented by the same amount

**Review by ambroise-arm — COMMENTED** [review]
It looks much better than my minimal implementation!

File: json-schema-check/validate_json_schemas.py (hunk 1 above)
Comment (xmfcx) [reply json-schema-check/validate_json_schemas.py]: Change: https://github.com/autowarefoundation/autoware-github-actions/pull/285/commits/2aec9b2c8a1b9f2bd42b40b948b1b7c88c1999d0

Example run: https://github.com/autowarefoundation/autoware.universe/actions/runs/7981506848/job/21813925896

Screenshot from 2024-02-21 15-08-54

This is what you've meant right?

Comment (ambroise-arm) [reply json-schema-check/validate_json_schemas.py]: yes, perfect

**Review by ambroise-arm — APPROVED** [review]

**Comment by xmfcx** [comment]
Thank you for the prompt review!

**Pull request merged**